### PR TITLE
chatgpt integration

### PR DESCRIPTION
**Pull request by ghost**
chatgpt integration:
added server implementation

client coming soon

### CODE REVIEW

File: MangoAPI.BusinessLogic/Models/Conversation.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace MangoAPI.BusinessLogic.Models;
+
+public sealed record Conversation(int LeftMessagesCount, string Request, bool IsResponseSucceed, string Response);
+
+public static class MessageAuthors
+{
+    public static readonly string User = "user";
```
Comment (kolosovpetro) [inline MangoAPI.BusinessLogic/Models/Conversation.cs]: `static readonly` strings can be `const`

File: MangoAPI.Domain/ChatGPTOptions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+namespace MangoAPI.Domain;
+
+public sealed class ChatGPTOptions
```
Comment (kolosovpetro) [inline MangoAPI.Domain/ChatGPTOptions.cs:3]: If here all properties are `init` only, then can this class be converted to record with primary constructor?

File: MangoAPI.Presentation/MangoAPI.Presentation.csproj [hunk 3]
```diff
@@ -12,6 +12,7 @@
     </PropertyGroup>
 
     <!-- Exclude the project from coverage report -->
+    <PropertyGroup Condition=" '$(RunConfiguration)' == 'MangoAPI' " />
```
Comment (kolosovpetro) [inline MangoAPI.Presentation/MangoAPI.Presentation.csproj]: What is the motivation for this change?

**Review by kolosovpetro — COMMENTED** [review]
Few questions to clear some spots in code.

**Comment by kolosovpetro** [comment]
Solves issue: https://github.com/MangoInstantMessenger/MangoMessengerAPI/issues/484

File: MangoAPI.Presentation/MangoAPI.Presentation.csproj (hunk 3 above)
Comment (ghost) [reply MangoAPI.Presentation/MangoAPI.Presentation.csproj]: Honestly I don't know how this line appeared, might was added by rider

File: MangoAPI.Domain/ChatGPTOptions.cs (hunk 2 above)
Comment (ghost) [reply MangoAPI.Domain/ChatGPTOptions.cs:3]: sure, but this is an option class(it will not be initialised manually) and don't think that get the purpose of doing this way

File: MangoAPI.BusinessLogic/Models/Conversation.cs (hunk 1 above)
Comment (ghost) [reply MangoAPI.BusinessLogic/Models/Conversation.cs]: made as a const props

**Review by kolosovpetro — COMMENTED** [review]
Looks good, thanks for fixes and clarifications

**Comment by kolosovpetro** [comment]
Btw about client-side update, should it be done in another PR or in same? Beacuse I don't know where to expect it and that's why haven't merged it yet

**Comment by ghost** [comment]
I thought to push it in another PR, For now it works fine from swaggerAPI [request](https://drive.google.com/file/d/1MXlROF3jafsEG8LjXAIhzM9_YosVqolg/view?usp=sharing)
[response](https://drive.google.com/file/d/1MXlROF3jafsEG8LjXAIhzM9_YosVqolg/view?usp=sharing)

**Comment by kolosovpetro** [comment]
> I thought to push it in another PR, For now it works fine from swaggerAPI [request](https://drive.google.com/file/d/1MXlROF3jafsEG8LjXAIhzM9_YosVqolg/view?usp=sharing) [response](https://drive.google.com/file/d/1MXlROF3jafsEG8LjXAIhzM9_YosVqolg/view?usp=sharing)

Cool I merge it then, thanks
